hamilton/driver.py

```diff
@@ -581,21 +582,30 @@ def _visualize_execution_helper(
         :param graphviz_kwargs:
         :return:
         """
-        # _final_vars = self._create_final_vars(final_vars)
-        nodes, user_nodes = fn_graph.get_upstream_nodes(final_vars, inputs)
+        nodes, user_nodes = fn_graph.get_upstream_nodes(final_vars, inputs, overrides)
         Driver.validate_inputs(fn_graph, adapter, user_nodes, inputs, nodes)
         node_modifiers = {fv: {graph.VisualizationNodeModifiers.IS_OUTPUT} for fv in final_vars}
         for user_node in user_nodes:
             if user_node.name not in node_modifiers:
                 node_modifiers[user_node.name] = set()
             node_modifiers[user_node.name].add(graph.VisualizationNodeModifiers.IS_USER_INPUT)
+        all_nodes = nodes | user_nodes
```

Comment:
will this work for all versions of python?

Reply:
otherwise why not use the `.union` ?

Reply:
Consistent with how we were doing it in another place, and yeah, this was a python 2 feature